### PR TITLE
feat(pagination): validating pagination inputs to receive first alone, last alone or the combination of {after, first} and {before,last}.

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal 
validate pagination for combination
- first
- first/after
- last
- last/after

## feedback on
- restricted to the max first/last limit to 100. this way, we can restrict large db read at once..

Jira ticket:
https://getpocket.atlassian.net/browse/INFRA-2

### CODE REVIEW

**Review by kschelonka — COMMENTED** [review]
Really thorough! Small change -- can we throw an Apollo UserInputError rather than a generic Error?

File: src/resolvers/utils.ts [hunk 1]
```diff
@@ -16,3 +17,60 @@ export function getSavedItemMapFromTags(tags) {
...
+  ) {
+    throw new Error('Please set either {after and first} or {before and last}');
+  }
+
+  if (pagination.before) {
+    const before = parseInt(
+      Buffer.from(pagination.before, 'base64').toString()
+    );
+    if (before < 0) {
+      throw new Error('invalid before cursor');
```
Comment (kkyeboah) [inline src/resolvers/utils.ts:40]: nit
```suggestion
      throw new Error('Invalid before cursor');
```

File: src/resolvers/utils.ts [hunk 2]
```diff
@@ -16,3 +17,60 @@ export function getSavedItemMapFromTags(tags) {
...
+
+    if (!pagination.last) {
+      pagination.last = defaultPage;
+    }
+  }
+
+  if (pagination.after) {
+    const after = parseInt(Buffer.from(pagination.after, 'base64').toString());
+    if (after < 0) {
+      throw new Error('invalid after cursor');
```
Comment (kkyeboah) [inline src/resolvers/utils.ts:51]: nit
```suggestion
      throw new Error('Invalid after cursor');
```

File: src/types/index.ts [hunk 3]
```diff
@@ -124,6 +124,7 @@ export type SavedItemUpsertInput = {
 };
 
 export const defaultPage = 30;
+export const maxPageSize = 100;
```
Comment (kkyeboah) [inline src/types/index.ts:127]: should these be coming from a `config` file instead of a `types` file?

**Review by kkyeboah — COMMENTED** [review]
Looks good to me. I had a few non-blocking nits

File: src/types/index.ts (hunk 3 above)
Comment (sri-kirsh) [reply src/types/index.ts]: moved to config

**Review by Herraj — APPROVED** [review]

**Pull request merged**